metrics/prometheus-metrics-endpoints-and-auto-discovery.md

```diff
@@ -106,6 +106,27 @@ datasources = /etc/grafana-datasources
 
 *NOTE:* This approach for making default dashboards available will be used until such a time as a more automated discovery of dashboards is determined. For example, is it possible to store the dashboard defininition with the Service that has metrics available e.g. Keycloak, and then make it show up in Grafana.
 
+#### Auto Discovery of Service Dashboard Definitions
+
+An approach to automated discovery of dashboard definitions is being investigated. The initial discussion of this approach can be seen [here](https://github.com/aerogear/proposals/pull/5#discussion_r157767612).
+
+The approach involves a sidecar service running inside the same pod as Grafana. This service would use the Kubernetes API to watch the namespace for Config Maps containing a Grafana dashboard definition (JSON). These Config Maps will be identified with an annotation. For example:
+
+```
+org.aerogear.grafana.dashboard: true
+```
+
+The sidecar service would feed the disovered dashboard JSON definitions into Grafana using an API call.
+
+The advantage is that the dashboard JSON would be bundled either with the service or with the service APB. This reduces the potential for version mismatches between a service and its associated Grafana dashboard. It also means downstream users could define their own dashboards for their services and have them auto-loaded into Grafana.
+
+The added cost to this approach is that a custom service might need to be built (or or an existing one will need extra development).
```

Comment:
While I love this approach, it's important that we don't underestimate this point. I have a feeling this would be a fairly significant undertaking in itself.